src/main/java/com/mongodb/hibernate/dialect/MongoDialectSettings.java

```diff
@@ -0,0 +1,293 @@
+                if (jdbcUrl instanceof String jdbcUrlText) {
+                    return parseConnectionString(JAKARTA_JDBC_URL, jdbcUrlText);
+                } else if (jdbcUrl instanceof ConnectionString jdbcUrlConnectionString) {
+                    return jdbcUrlConnectionString;
+                } else {
+                    throw Exceptions.unsupportedType(JAKARTA_JDBC_URL, jdbcUrl, String.class, ConnectionString.class);
+                }
+            }
+
+            static @Nullable Boolean getAutoCommit(Map<String, Object> configProperties) {
+                var autoCommit = configProperties.get(AUTOCOMMIT);
+                if (autoCommit == null) {
+                    return null;
+                }
+                if (autoCommit instanceof String autoCommitText) {
+                    return parseBooleanString(AUTOCOMMIT, autoCommitText);
+                } else if (autoCommit instanceof Boolean autoCommitBoolean) {
+                    return autoCommitBoolean;
+                } else {
+                    throw Exceptions.unsupportedType(AUTOCOMMIT, autoCommit, String.class, Boolean.class);
+                }
+            }
+
+            private static ConnectionString parseConnectionString(String propertyName, String propertyValue) {
+                try {
+                    return new ConnectionString(propertyValue);
+                } catch (RuntimeException e) {
+                    throw Exceptions.failedToParse(propertyName, propertyValue, ConnectionString.class);
+                }
+            }
+
+            private static boolean parseBooleanString(String propertyName, String propertyValue) {
+                return switch (propertyValue) {
+                    case "true" -> true;
+                    case "false" -> false;
+                    default -> throw Exceptions.failedToParse(propertyName, propertyValue, boolean.class);
+                };
+            }
+
+            private static final class Exceptions {
+                static RuntimeException unsupportedType(
+                        String propertyName, Object propertyValue, Type... expectedTypes) {
+                    return new RuntimeException(format(
+                            "Type %s of configuration property [%s] with value [%s] must be one of %s",
+                            propertyValue.getClass().getTypeName(),
+                            propertyName,
+                            propertyValue,
+                            Arrays.stream(expectedTypes).map(Type::getTypeName).collect(Collectors.joining(", "))));
+                }
+
+                static RuntimeException failedToParse(String propertyName, String propertyValue, Type type) {
+                    return new RuntimeException(format(
+                            "Failed to get %s from configuration property [%s] with value [%s]",
+                            type.getTypeName(), propertyName, propertyValue));
+                }
+            }
+        }
+    }
```

Comment:
Hibernate has internal configuration parsing helper class: https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/jpa/internal/util/ConfigurationHelper.java.

Seems throwing `IllegalArgumentException` is ok. Throwing `HibernateException` is not a hard rule and I think its wrapping will be done by Hibernate internally.

Reply:
I don't think we should copy that class from Hibernate ORM to our code.